chef/cookbooks/neutron/recipes/l3.rb

```diff
@@ -175,7 +175,9 @@
 # and because it listens on this specific IP address only (so we don't want to use a name
 # that could resolve to 127.0.0.1).
 metadata_host = CrowbarHelper.get_host_for_admin_url(nova, (nova[:nova][:ha][:enabled] rescue false))
-metadata_port = "8775"
+metadata_port = nova[:nova][:ports][:metadata] rescue 8775
+metadata_protocol = (nova[:nova][:ssl][:enabled] ? "https" : "http") rescue "http"
```

Comment:
The comment above this code should be updated. Do we also need to switch ```metadata_host``` do you a fqdn when SSL is enabled?